### DISCUSSION TITLE
Shelly EM - Einzelne Kanäle ansprechen

### DISCUSSION THREAD

**Discussion by GiK90** [camueller/SmartApplianceEnabler#243] [Q&A]
Hallo zusammen!

Erstmal vielen Dank für dieses Klasse Programm!

Habe jetzt einen [Shelly EM](https://shop.shelly.cloud/shelly-em-50a-clamp-wifi-smart-home-automation-1#139) angeschafft und würde diesen beiden Kanäle gerne auf einzelne Geräte im SunnyPortal auslesen lassen. Meine aktuelle Version 1.6.18

Die Ausgabe unter ip/status bekomme ich aktuell so:

"emeters":
[
{"power":0.00,"reactive":0.00,"pf":0.00,"voltage":224.80,"is_valid":true,"total":12.3,"total_returned":0.6},{"power":2.69,"reactive":0.00,"pf":0.00,"voltage":224.80,"is_valid":true,"total":1512.8,"total_returned":448.9}
],
"update":{"status":"idle","has_update":false,"new_version":"20220209-094734/v1.11.8-g8c7bb8d",
"old_version":"20220209-094734/v1.11.8g8c7bb8d"},
"ram_total":50256,"ram_free":35184,"fs_size":233681,"fs_free":157879,"uptime":190304}

Wie bekomme ich nun die einzelnen "Kanäle"?

Und 1 Kanal von den beiden ist 1 Phase meiner Sauna. Diesen Wert müsste ich mit 3 multiplizieren, geht das über das Regex?
Original wäre ja so:
"power":([\d]+)

Wenn ich den Wert mit 3 multiplizieren will geht das irgendwie so dann?
"power":(([\d]+)*3)

Vielen Dank!

**Comment by camueller** [comment]
Du kannst als Format JSON und den entsprechenden Pfad angeben - [siehe Doku](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/HttpMeter_DE.md). Multiplizieren kannst Du über den Umrechnungsfaktor.

**Reply by GiK90** [reply]
Danke für die schnelle Antwort. Leider kann ich hier nichts wegen des Arrays finden...

  "emeters": [
    {
      "power": 0,
      "reactive": 0,
      "pf": 0,
      "voltage": 225.32,
      "is_valid": true,
      "total": 14.9,
      "total_returned": 0.6
    },
    {
      **"power": 11.32,**
      "reactive": -23.91,
      "pf": -0.43,
      "voltage": 225.32,
      "is_valid": true,
      "total": 1526.6,
      "total_returned": 448.9
    }
  ]

Ich möchte den 2. Power-Wert auslesen (11.32 Watt)
So funktioniert es leider nicht:

![image](https://user-images.githubusercontent.com/62926420/161611883-6411e822-bc63-4e79-ab1a-3ea7f95492ba.png)

**Reply by GiK90** [reply]
Ah! Jetzt weiß ich, wo der Fehler liegt. :)
So funktioniert es. Danke

![image](https://user-images.githubusercontent.com/62926420/161614186-14575834-c86d-4d1f-9ba7-9198619b3059.png)
